### DISCUSSION TITLE
iOS support

### DISCUSSION THREAD

**Discussion by mmazzoli-dev** [pkdawson/imgui-godot#42] [Q&A] [answered]
First off this is amazing and thanks so much for it. I recently tried an iOS export in the new Godot 4.2 beta that brings back (experimental) C#/iOS support and an error is thrown in XCode using this library, due to iOS disallowing code generation (I think it was Reflection.Emit specifically). I don't know much about the inner workings of imgui-godot but just thought I'd ask to see if there was a way around this. Thanks :)

**Comment by pkdawson** [comment, marked as answer]
The problem is that ImGui.NET doesn't support iOS (or Android), so even if I fix this issue it would still either fail to build or crash at runtime.

I'll need to figure out a way of excluding ImGui code for mobile export. Unfortunately it will probably also require users to wrap all their ImGui calls in `#if !GODOT_MOBILE` or something, but I'll do some research.

**Reply by mmazzoli-dev** [reply]
Ah fair enough, I figured ImGui.NET might work with single touch emulating the mouse but if that's not the case no matter.

Yeah I can't think of a tidy way from your side to exclude the library from mobile, at least until godot gets something similar to Unity's platform specific assemblies. It's straight forward enough however as a user, you can specify to omit the imgui-godot folder in a mobile platform's export settings, then just disable the plugin's autoload and manually load the ImGuiLayer.tscn #if !GODOT_MOBILE.

Anyway thanks again for your work on this, it's been supremely useful as a debug toolkit for my game.

**Comment by pkdawson** [comment]
Theoretically mobile support would be possible, since Godot takes care of all the hard parts of platform support. But ImGui.NET simply doesn't build the native cimgui library (which is essentially platform-agnostic) for them, just Windows/Mac/Linux. That might be another possible solution to the problem though.

But for now yeah I think I'll just disable imgui-godot on mobile, and add the necessary documentation.
